src/Lean/Parser/Term.lean

```diff
@@ -448,6 +448,10 @@ def withAnonymousAntiquot := leading_parser
 @[builtin_term_parser] def «trailing_parser» := leading_parser:leadPrec
   "trailing_parser" >> optExprPrecedence >> optExprPrecedence >> ppSpace >> termParser
 
+/-- Marks a function argument of an `@[extern]`ed function as borrowed.
+
+See https://lean-lang.org/lean4/doc/dev/ffi.html#borrowing;
```

Comment:
The linked section isn't all that long, perhaps it should be summarized here instead?

Reply:
My thinking was mainly that the linked section had a lot of remarks about the current state of development, and so duplicating the information increased the chance of things going out of sync. I can duplicate it here verbatim if you'd prefer; though if you would prefer to reword it, perhaps it's best for you to push directly to this PR.

Reply:
I think it should be explained in more detail inline than this. The comment as is adds no information besides the word "borrowed". I think the important information for users is:

* This is an optimization hint and does not change the semantics of the code
* This is only relevant for @[extern] functions and does nothing unless it appears in a parameter type (AFAIK there is no borrowing annotation for return types, even though `b_obj_res` exists)
* This changes the ABI for the function to take an argument as "borrowed", meaning that it does not consume the value (or lower the RC value by 1 if it is shared)

Reply:
I think this makes sense - I used that as a starting point for a suggested revision below. What do you think?

Reply:
> The comment as is adds no information besides the word "borrowed".

This is the difference between hovering over the symbol in vscode showing nothing, and hovering over it showing a word that can be searched on Zulip.

@david-christiansen, would you be happy to adopt this PR? I'm unsure how to address your concerns simultaneously with Mac and Mario's, and would at this point be happy with any non-empty docstring here.